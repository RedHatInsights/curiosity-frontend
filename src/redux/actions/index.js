import * as rhelActions from './rhelActions';
import * as userActions from './userActions';

const actions = {
  rhel: rhelActions,
  user: userActions
};

const reduxActions = { ...actions };

export { reduxActions as default, reduxActions, rhelActions, userActions };
