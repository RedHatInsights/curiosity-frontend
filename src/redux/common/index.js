import { reduxHelpers } from './reduxHelpers';

export { reduxHelpers as default, reduxHelpers };
