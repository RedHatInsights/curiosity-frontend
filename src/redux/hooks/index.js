import { reactReduxHooks } from './useReactRedux';

const storeHooks = {
  reactRedux: reactReduxHooks
};

export { storeHooks as default, storeHooks, reactReduxHooks };
