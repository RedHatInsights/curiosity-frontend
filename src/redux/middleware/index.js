import { createLogger } from 'redux-logger';
import promiseMiddleware from 'redux-promise-middleware';
import thunkMiddleware from 'redux-thunk';

const reduxMiddleware = [thunkMiddleware, promiseMiddleware];

if (process.env.NODE_ENV !== 'production' && process.env.REACT_APP_DEBUG_MIDDLEWARE === 'true') {
  reduxMiddleware.push(createLogger());
}

export { reduxMiddleware as default, reduxMiddleware, createLogger, promiseMiddleware, thunkMiddleware };
