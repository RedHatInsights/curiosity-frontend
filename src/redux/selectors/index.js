const reduxSelectors = {};

export { reduxSelectors as default, reduxSelectors };
