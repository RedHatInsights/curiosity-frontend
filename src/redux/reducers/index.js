import { combineReducers } from 'redux';
import { notifications } from '@redhat-cloud-services/frontend-components-notifications';
import graphReducer from './graphReducer';
import inventoryReducer from './inventoryReducer';
import viewReducer from './viewReducer';
import userReducer from './userReducer';

const reducers = {
  notifications,
  graph: graphReducer,
  inventory: inventoryReducer,
  user: userReducer,
  view: viewReducer
};

const reduxReducers = combineReducers(reducers);

export { reduxReducers as default, reduxReducers, graphReducer, inventoryReducer, userReducer, viewReducer };
