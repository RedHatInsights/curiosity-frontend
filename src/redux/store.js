import { createStore, applyMiddleware } from 'redux';
import { reduxMiddleware } from './middleware';
import { reduxReducers } from './reducers';

const store = createStore(reduxReducers, applyMiddleware(...reduxMiddleware));

export { store as default, store };
