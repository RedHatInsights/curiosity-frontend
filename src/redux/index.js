import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { withTranslation } from 'react-i18next';
import { store } from './store';
import { reduxActions } from './actions';
import { reduxReducers } from './reducers';
import { reduxSelectors } from './selectors';
import { reduxTypes } from './types';
import { helpers } from '../common/helpers';

const connectTranslate = (mapStateToProps, mapDispatchToProps) => component =>
  connect(
    mapStateToProps,
    mapDispatchToProps
  )((!helpers.TEST_MODE && withTranslation()(component)) || component);

const connectRouterTranslate = (mapStateToProps, mapDispatchToProps) => component =>
  withRouter(connectTranslate(mapStateToProps, mapDispatchToProps)(component));

const connectRouter = (mapStateToProps, mapDispatchToProps) => component =>
  withRouter(
    connect(
      mapStateToProps,
      mapDispatchToProps
    )(component)
  );

export {
  connect,
  connectRouter,
  connectRouterTranslate,
  connectTranslate,
  reduxActions,
  reduxReducers,
  reduxSelectors,
  reduxTypes,
  store
};
