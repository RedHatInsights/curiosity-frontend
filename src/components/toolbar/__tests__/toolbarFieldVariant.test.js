import React from 'react';
import { ToolbarFieldVariant, useToolbarFieldOptions, useOnSelect } from '../toolbarFieldVariant';
import { store } from '../../../redux/store';
import {
  RHSM_API_PATH_PRODUCT_VARIANT_RHEL_TYPES as RHEL_TYPES,
  RHSM_API_QUERY_SET_TYPES
} from '../../../services/rhsm/rhsmConstants';

describe('ToolbarFieldVariant Component', () => {
  let mockDispatch;

  beforeEach(() => {
    mockDispatch = jest.spyOn(store, 'dispatch').mockImplementation((type, data) => ({ type, data }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render a basic component', async () => {
    const props = {
      useProductToolbarQuery: () => ({ [RHSM_API_QUERY_SET_TYPES.VARIANT]: RHEL_TYPES.RHEL_X86 })
    };
    const component = await shallowComponent(<ToolbarFieldVariant {...props} />);

    expect(component).toMatchSnapshot('basic');
  });

  it('should generate select options', async () => {
    const { result: toolbarFieldOptions } = await renderHook(() =>
      useToolbarFieldOptions({
        useProduct: () => ({
          productVariants: ['lorem', 'ipsum', 'dolor', 'sit', '']
        })
      })
    );

    expect(toolbarFieldOptions).toMatchSnapshot('toolbarFieldOptions');
  });

  it('should handle updating through redux state with component', () => {
    const props = {
      useToolbarFieldOptions: () => [
        {
          selected: false,
          title: 't(curiosity-toolbar.label_variant, {"context":"ipsum"})',
          value: 'ipsum'
        },
        {
          selected: false,
          title: 't(curiosity-toolbar.label_variant, {"context":"lorem"})',
          value: 'lorem'
        }
      ]
    };

    const component = renderComponent(<ToolbarFieldVariant {...props} />);
    const input = component.find('button');
    component.fireEvent.click(input);

    const inputMenuItem = component.find('button.pf-c-select__menu-item');
    component.fireEvent.click(inputMenuItem);

    expect(mockDispatch.mock.calls).toMatchSnapshot('dispatch, component');
  });

  it('should handle updating through redux state with hook', () => {
    const options = {
      useProduct: () => ({ viewId: 'loremIpsum' })
    };

    const onSelect = useOnSelect(options);
    onSelect({
      value: 'dolor sit'
    });

    expect(mockDispatch.mock.calls).toMatchSnapshot('dispatch, hook');
  });
});
