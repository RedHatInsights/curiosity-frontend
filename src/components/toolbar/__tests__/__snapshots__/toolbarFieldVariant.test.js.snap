// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarFieldVariant Component should generate select options: toolbarFieldOptions 1`] = `
[
  {
    "selected": false,
    "title": "t(curiosity-toolbar.label_variant, {"context":"lorem"})",
    "value": "lorem",
  },
  {
    "selected": false,
    "title": "t(curiosity-toolbar.label_variant, {"context":"ipsum"})",
    "value": "ipsum",
  },
  {
    "selected": false,
    "title": "t(curiosity-toolbar.label_variant, {"context":"dolor"})",
    "value": "dolor",
  },
  {
    "selected": false,
    "title": "t(curiosity-toolbar.label_variant, {"context":"sit"})",
    "value": "sit",
  },
  {
    "selected": false,
    "title": "t(curiosity-toolbar.label_variant, {"context":"none"})",
    "value": "",
  },
]
`;

exports[`ToolbarFieldVariant Component should handle updating through redux state with component: dispatch, component 1`] = `
[
  [
    [
      {
        "type": "SET_QUERY_RESET_INVENTORY_LIST",
        "viewId": undefined,
      },
      {
        "filter": "variant",
        "type": "SET_QUERY",
        "value": "ipsum",
        "viewId": undefined,
      },
    ],
  ],
]
`;

exports[`ToolbarFieldVariant Component should handle updating through redux state with hook: dispatch, hook 1`] = `
[
  [
    [
      {
        "type": "SET_QUERY_RESET_INVENTORY_LIST",
        "viewId": "loremIpsum",
      },
      {
        "filter": "variant",
        "type": "SET_QUERY",
        "value": "dolor sit",
        "viewId": "loremIpsum",
      },
    ],
  ],
]
`;

exports[`ToolbarFieldVariant Component should render a basic component: basic 1`] = `
<Select
  aria-label="t(curiosity-toolbar.placeholder, {"context":"variant"})"
  ariaLabel="Select option"
  className=""
  data-test="toolbarFieldVariant"
  direction="down"
  id="generatedid-"
  isDisabled={false}
  isDropdownButton={false}
  isFlipEnabled={false}
  isInline={true}
  isToggleText={true}
  maxHeight={null}
  name={null}
  onSelect={[Function]}
  onSplitButton={[Function]}
  options={[]}
  placeholder="t(curiosity-toolbar.placeholder, {"context":"variant"})"
  position="left"
  selectedOptions="RHEL for x86"
  splitButtonVariant={null}
  toggleIcon={null}
  variant="single"
/>
`;
